src/model/engines/engine_base.nit

```diff
@@ -53,5 +76,79 @@ class Engine
 	fun compile(submission: Submission): Bool is abstract
 
 	# Run `test` for `submission`
-	fun run_test(submission: Submission, test: TestCase): TestResult is abstract
+	fun run_test(submission: Submission, test: TestCase): TestResult do
+		var res = new TestResult(test, submission)
+
+		var tdir = "{submission.mission.id}_{submission.player.id}"
```

Comment:
the point of `tdir` is to have a distinct subdirectory per test so that:

* a test does not pollute an other one
* the administration or professor can inspect each directory

In #82 , I changed it to

~~~nit
var tdir = "test{program.results.length + 1}"
~~~

please keep the change